tools/wt_to_mdb_bson.py

```diff
@@ -53,24 +143,67 @@ def decode_data_section(mdb_file_contents, data_index):
         obj = bson.decode_all(byt)[0]
 
         print('Key:\t%s' % key)
-        print('Value:\n\t%s' % (pprint.pformat(obj, indent=1).replace('\n', '\n\t'),))
+        print('Value:\n\t%s' % (print_bson(obj),))
 
-def dump_mdb_file(wtpath, filename):
+# Convert the output of ./wt -r dump -x to bson.
+def wt_dump_to_bson(wt_output):
     # Dump the MongoDB file into hex format.
-    mdb_hex = subprocess.check_output([wtpath, "dump", "-x", "file:" + filename], universal_newlines=True)
-
-    mdb_file_contents = mdb_hex.splitlines()
+    mdb_file_contents = wt_output
     data_index = find_data_section(mdb_file_contents)
     if data_index > 0:
         decode_data_section(mdb_file_contents, data_index)
     else:
         print("Error: No data section was found in the file.")
-        exit()        
+        exit()
+
+# Call the wt util if required.
+def execute_wt(mode, wtpath, filename):
+    if mode == Mode.DUMP:
+        return subprocess.check_output(
+            [wtpath, "-r", "dump", "-x", "file:" + filename], universal_newlines=True).splitlines()
+    elif mode == Mode.VERIFY:
+        return subprocess.check_output(
+            [wtpath, "-r", "verify", "-d", "dump_pages", "file:" + filename], universal_newlines=True).splitlines()
+    else:
+        return subprocess.check_output(
+            [wtpath, "-r", "printlog", "-u", "-x"], universal_newlines=True).splitlines()
+
+def main():
+    if len(sys.argv) < 3:
+        util_usage()
+        exit()
+
+    if sys.argv[1] != '-m':
+        print('A mode must be specified with -m.')
+        util_usage()
 
-if len(sys.argv) != 3:
-    util_usage()
-    exit()
+    mode_str = sys.argv[2]
+    if mode_str == 'dump':
+        mode = Mode.DUMP
+    elif mode_str == 'verify':
+        mode = Mode.VERIFY
+    elif mode_str == 'printlog':
+        mode = Mode.PRINTLOG
+    else:
+        print('Invalid mode specified.')
+        util_usage()
+
+    # Does the user plan on passing wt's location and a file?
+    if len(sys.argv) > 3:
+        if sys.argv[3] != '-i':
+            print('Invalid option specified.')
+            util_usage()
+        wt_output = execute_wt(mode, sys.argv[4], sys.argv[5])
```

Comment:
`./wt_to_mdb_bson -m printlog -f ./wt` doesn't require a .wt file so `sys.argv[5]` can be an out of bounds access here.

Given we already know `mode` I'd just add a
`wt_file = None if mode == Mode.PRINTLOG else sys.argv[5]` and pass that into `execute_wt`

Reply:
73875cd5e